### PR TITLE
Jonmv/differentiate between prod and non prod instances in rest api

### PR DESCRIPTION
**Pull request by jonmv**
@freva please review and merge.

### CODE REVIEW

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/responses/jobs.json [hunk 1]
```diff
@@ -335,7 +335,7 @@
       "runs": [
         {
           "id": 1,
-          "url": "http://localhost:8080/application/v4/tenant/tenant1/application/application1/instance/instance1/job/run/run 1 of dev-us-east-1 for tenant1.application1.instance1",
+          "url": "http://localhost:8080/application/v4/tenant/tenant1/application/application1/instance/instance1/job?/run/run 1 of dev-us-east-1 for tenant1.application1.instance1",
```
Comment (freva) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/responses/jobs.json]: ?

**Review by freva — COMMENTED** [review]
Is there anything that prevents a dev deployment to a production instance?

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/responses/jobs.json (hunk 1 above)
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/responses/jobs.json]: Oops.

**Comment by jonmv** [comment]
Yes, there is a check that manual deployments are not made to an environment that is not manually deployed.

**Comment by freva** [comment]
Yeah, the environment part is fine, but I can have a production deployment on instance `default` and then deploy directly to `dev` also for instance `default` right? In that case, this API will return `default` as a production instance?

**Comment by jonmv** [comment]
Yes. Removed that part, as it wasn't too useful.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/responses/jobs.json (hunk 1 above)
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/responses/jobs.json]: The `?` is not the worst part — read the rest of the line!

**Review by freva — APPROVED** [review]

**Pull request merged**